src/pages/AddData.tsx

```diff
@@ -41,10 +49,20 @@ const BulkUploader = ({ onBulkUpload }: BulkUploaderProps) => {
 
 		await uploadUsingPresignedPost(file, presignedPost);
 
-		await registerBulkUpload({
-			fileName: file.name,
-			sourceKey: presignedPost.fields.key,
-		});
+		let systemSource: Source | undefined;
+		if (sources?.entries) {
+			systemSource = getSystemSource(sources.entries, SYSTEM_SOURCE_SHORT_CODE);
+		}
+
+		if (systemSource?.id) {
+			await registerBulkUpload({
+				fileName: file.name,
+				sourceKey: presignedPost.fields.key,
+				sourceId: systemSource?.id,
+			});
+		} else {
+			throw new Error('No System Source Available');
```

Comment:
Good call throwing an `Error` here. I wonder how it surfaces and if the message could be more explicit, like `No System Source with short code = yada found at [url used]`.